core/drivers/amd/ps_gpio_driver.c

```diff
@@ -0,0 +1,246 @@
+		offset = DATA_MSW_OFFSET(bank);
+	}
+
+	/* Explicitly compare the enum value */
+	if (level == GPIO_LEVEL_HIGH)
+		level = 1;
+	else
+		level = 0;
+
+	level = ~BIT32(pin) & (SHIFT_U32(level, pin) | GPIO_UPPER_MASK);
+
+	io_write32(ps->vbase + offset, level);
+}
+
+static enum gpio_dir ps_gpio_get_dir(struct gpio_chip *chip,
+				     unsigned int gpio_pin)
+{
+	uint32_t bank = 0;
+	uint32_t pin = 0;
+	struct amd_gpio_info *ps = container_of(chip, struct amd_gpio_info,
+						chip);
+
+	get_bank_and_pin(ps->bdata, gpio_pin, &bank, &pin);
+
+	if (io_read32(ps->vbase + DIRM_OFFSET(bank)) & BIT(pin))
+		return GPIO_DIR_OUT;
+
+	return GPIO_DIR_IN;
+}
+
+static void ps_gpio_set_dir(struct gpio_chip *chip,
+			    unsigned int gpio_pin,
+			    enum gpio_dir direction)
+{
+	uint32_t bank = 0;
+	uint32_t pin = 0;
+	uint32_t regval = 0;
+	struct amd_gpio_info *ps = container_of(chip, struct amd_gpio_info,
+						chip);
+
+	get_bank_and_pin(ps->bdata, gpio_pin, &bank, &pin);
+
+	if (direction == GPIO_DIR_OUT) {
+		/* set the GPIO pin as output */
+		regval = io_read32(ps->vbase + DIRM_OFFSET(bank));
+		regval |= BIT(pin);
+		io_write32(ps->vbase + DIRM_OFFSET(bank), regval);
+
+		/* configure the output enable reg for the pin */
+		regval = io_read32(ps->vbase + OUTEN_OFFSET(bank));
+		regval |= BIT(pin);
+		io_write32(ps->vbase + OUTEN_OFFSET(bank), regval);
+
+		/* set the state of the pin */
+		ps_gpio_set_value(chip, gpio_pin, GPIO_LEVEL_LOW);
+	} else {
+		regval = io_read32(ps->vbase + DIRM_OFFSET(bank));
+		regval &= ~BIT(pin);
+		io_write32(ps->vbase + DIRM_OFFSET(bank), regval);
+	}
+}
+
+static enum gpio_interrupt ps_gpio_get_intr(struct gpio_chip *chip,
+					    unsigned int gpio_pin)
+{
+	uint32_t bank = 0;
+	uint32_t pin = 0;
+	struct amd_gpio_info *ps = container_of(chip, struct amd_gpio_info,
+						chip);
+
+	get_bank_and_pin(ps->bdata, gpio_pin, &bank, &pin);
+
+	if ((io_read32(ps->vbase + INT_MASK_OFFSET(bank)) & BIT(pin)))
+		return GPIO_INTERRUPT_DISABLE;
+
+	return GPIO_INTERRUPT_ENABLE;
+}
+
+static void ps_gpio_set_intr(struct gpio_chip *chip,
+			     unsigned int gpio_pin,
+			     enum gpio_interrupt interrupt)
+{
+	uint32_t bank = 0;
+	uint32_t pin = 0;
+	uint32_t offset = 0;
+	uint32_t mask = 0;
+	uint32_t regval = 0;
+	struct amd_gpio_info *ps = container_of(chip, struct amd_gpio_info,
+						chip);
+
+	get_bank_and_pin(ps->bdata, gpio_pin, &bank, &pin);
+
+	mask = io_read32(ps->vbase + INT_MASK_OFFSET(bank)) & BIT(pin);
+
+	/* mask = 1 --> Interrupt is masked/disabled.
+	 * mask = 0 --> Interrupt is un-masked/enabled.
+	 */
+	if (mask && interrupt == GPIO_INTERRUPT_ENABLE) {
+		offset = INT_EN_OFFSET(bank);
+	} else if (!mask && interrupt == GPIO_INTERRUPT_DISABLE) {
+		offset = INT_DIS_OFFSET(bank);
+	} else {
+		IMSG("No change, interrupt already %s",
+		     interrupt ? "Enabled" : "Disabled");
+		return;
+	}
+
+	regval = io_read32(ps->vbase + offset);
+	regval &= BIT(pin);
+	io_write32(ps->vbase + offset, regval);
```

Comment:
It is intended to clear all bits but  `BIT(pin)` that is left to its current value? 
Shouldn't lin 173 rather be `|=` (in which case `io_setbits32(ps->vbase + offset, BIT(pin))` would do the job).